### PR TITLE
refactor: templating stac-validate workflow TDE-1136

### PR DESCRIPTION
**Pull request by paulfouquet**
#### Motivation

`stac-validate` is used in many places, it should be easier to maintain and reuse if it was a "template".

#### Modification

Create a `stac-validate` template so it can be re-use by workflows

#### Checklist

_If not applicable, provide explanation of why._

- [ ] Tests updated
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: templates/argo-tasks/stac-validate.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        resources:
+          requests:
+            cpu: 15000m
+            memory: 7.8Gi
+        command: [node, /app/index.js]
+        env:
+          - name: AWS_ROLE_CONFIG_PATH
+            value: s3://linz-bucket-config/config.json
+        args:
+          [
```
Comment (blacha) [inline templates/argo-tasks/stac-validate.yml:50]: NIT: do we have a preference on `[ .. ]` vs `- - -` style for arrays?

File: templates/argo-tasks/stac-validate.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+          - name: version
+            description: container version to use
+            default: 'v3'
+
+      container:
+        image: '019359803926.dkr.ecr.ap-southeast-2.amazonaws.com/argo-tasks:{{=sprig.trim(inputs.parameters.version)}}'
+        resources:
+          requests:
+            cpu: 15000m
```
Comment (blacha) [inline templates/argo-tasks/stac-validate.yml:43]: Is there anyway to scale this differently if `--checksum` is enabled, we do not need 15CPUs to validate a few JSON files?

**Review by l0b0 — DISMISSED** [review]

File: templates/argo-tasks/stac-validate.yml (hunk 2 above)
Comment (l0b0) [reply templates/argo-tasks/stac-validate.yml:43]: We should probably do that in a separate ticket, rather than changing two unrelated things in this one.

Comment (paulfouquet) [reply templates/argo-tasks/stac-validate.yml:43]: We should probably do it in this change since we are making it a generic template.

**Review by l0b0 — DISMISSED** [review]

File: templates/argo-tasks/stac-validate.yml (hunk 2 above)
Comment (paulfouquet) [reply templates/argo-tasks/stac-validate.yml:43]: @blacha It does not seem possible to [configure the pod resources using parameters](https://github.com/argoproj/argo-workflows/issues/7839). I would not want [to use `podSpecPatch`](https://github.com/argoproj/argo-workflows/issues/3058#issuecomment-631206170) in this pull request.

**Review by blacha — APPROVED** [review]

**Review by MDavidson17 — APPROVED** [review]

**Review by MDavidson17 — APPROVED** [review]

**Review by l0b0 — APPROVED** [review]

**Pull request merged**